### PR TITLE
[develop]: Updated exregional_get_obs scripts

### LINKED ISSUE #823: exregional_get_obs scripts for verification do not work for recent dates
## Expected behavior
When executing the workflow for an experiment that requires pulling verification observation files from the NOAA HPSS, it should execute successfully.

## Current behavior
The exregional_get_obs scripts have not been updated in over a year. In that time, the tar file names on the NOAA HPSS have been updated; however, those changes have not been reflected in the exregional_get_obs scripts. Therefore, more recent dates fail when you try and pull them from the NOAA HPSS.

## Machines affected
Tasks for pulling recent observation files needed for verification will fail on any platform that is trying to access the NOAA HPSS.

## Steps To Reproduce
Setup an experiment to run a 6-h forecast including pulling the observation files from the NOAA HPSS. I ran 20230515 00 UTC, and it failed. 

## Detailed Description of Fix (optional)
Modify exregional_get_obs_ccpa.sh, exregional_get_obs_mrms.sh, and exregional_get_obs_ndas.sh to update date ranges of file names on the NOAA HPSS. A longer term fix would to include verification obs similar to exregional_get_da_obs.sh.

### PR DESCRIPTION
**Pull request by michelleharrold**
## DESCRIPTION OF CHANGES:
This PR addressed Issue #823. For more recent dates (June 2022 and beyond), the exregional_get_obs scripts fail because the tar files names on the NOAA HPSS were updated but not updated accordingly in the pull scripts. A simple fix of modifying the logic to account for mapping the valid date to the tar file name have been updated in `scripts/exregional_get_obs_ccpa.sh`, `scripts/exregional_get_obs_mrms.sh`, and `scripts/exregional_get_obs_ndas.sh`.

As a note: To avoid having to continually update file name changes/location on the HPSS, we will be looking to leverage some of the capabilities being used for pulling DA observations and overhauling the verification observation pull scripts. @mkavulich will be looking into this soon. 

### Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## TESTS CONDUCTED: 
The modifications were tested on Hera only. Several dates were tested (20220619, 20230515) for the FV3_HRRR suite over RRFS_CONUS_25km domain. 24-h simulations were run and included pulling obs from NOAA HPSS and running verification.

- [X] hera.intel
- [ ] orion.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [ ] gaea.intel
- [ ] jet.intel
- [ ] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [ ] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## DEPENDENCIES:
None.

## DOCUMENTATION:
N/A.

## ISSUE: 
Fixes issue mentioned in #823.

## CHECKLIST

- [ ] My code follows the style guidelines in the Contributor's Guide
- [ ] I have performed a self-review of my own code using the Code Reviewer's Guide
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

## LABELS (optional): 

A Code Manager needs to add the following labels to this PR: 
- [ ] Work In Progress
- [ ] bug
- [ ] enhancement
- [ ] documentation
- [ ] release
- [ ] high priority
- [ ] run_ci
- [ ] run_we2e_fundamental_tests
- [ ] run_we2e_comprehensive_tests
- [ ] Needs Cheyenne test 
- [ ] Needs Jet test 
- [ ] Needs Hera test 
- [ ] Needs Orion test 
- [ ] help wanted

### CODE REVIEW

File: scripts/exregional_get_obs_ccpa.sh [hunk 1]
```diff
@@ -79,7 +79,7 @@ while [[ ${current_fcst} -le ${fcst_length} ]]; do
   vdd=`echo ${vdate} | cut -c7-8`    # day (DD) of valid time
   vhh=`echo ${vdate} | cut -c9-10`       # forecast hour (HH)
 
-  vhh_noZero=$(expr ${vhh} + 0)
+  vhh_noZero=$( printf "%d" "${vhh#0}" )
```
Comment (gsketefian) [inline scripts/exregional_get_obs_ccpa.sh]: @michelleharrold Is the `#0` at the end something I suggested in our meeting?  If so, sorry, it's not correct!  Scrap that and change to `$((10#${vhh}))` to force `${vhh}` to be interpreted as a decimal number.  And please run a test!

Comment (michelleharrold) [reply scripts/exregional_get_obs_ccpa.sh]: @gsketefian -- what you originally suggested worked...until I tested on forecast hour 08 and 09, and then I got the dreaded octet error. So I had to modify it. I will go ahead and change to what you suggested and run it!

File: scripts/exregional_get_obs_mrms.sh [hunk 2]
```diff
@@ -150,7 +154,7 @@ Did you forget to run \"module load hpss\"?\
     fi
 
   else
-    echo "mrms_file exists: \"$mrms_proc/${vyyyymmdd}/${field_base_name}${level}${vyyyy}${vmm}${vdd}-${vhh}0000.grib2\" No work to be done."
+    echo "mrms_file exists: \"$mrms_proc/${vyyyymmdd}/${field_base_name}${level}${vyyyy}${vmm}${vdd}-${}0000.grib2\" No work to be done."
```
Comment (MichaelLueken) [inline scripts/exregional_get_obs_mrms.sh]: @michelleharrold While attempting to run the get_obs_mrms task on Hera, I'm seeing the following error message:

```
MRMS FILE:/scratch2/NAGAPE/epic/Michael.Lueken/UFS_SRW_data/develop/obs_data/mrms/proc/../proc/20200615/MergedReflectivityQCComposite_00.50_20200615-190000.grib2
/scratch2/NAGAPE/epic/Michael.Lueken/ufs-srweather-app/scripts/exregional_get_obs_mrms.sh: line 157: mrms_file exists: "$mrms_proc/${vyyyymmdd}/${field_base_name}${level}${vyyyy}${vmm}${vdd}-${}0000.grib2" No work to be done.: bad substitution
```
Is there a reason that you removed vhh from the bracket on line 157?  This behavior has been seen for several different dates.

When I added vhh back in, the task successfully completes and the verification test runs through to completion.

Comment (michelleharrold) [reply scripts/exregional_get_obs_mrms.sh]: @MichaelLueken -- that must an error on my part (likely accidentally removed when I was searching for vhh due to the suggestion from @gsketefian to correct how we set vhh_noZero). I will fix that and push the mods. Thanks for catching that!

Comment (MichaelLueken) [reply scripts/exregional_get_obs_mrms.sh]: Thanks for the modification, @michelleharrold!  Resolving conversation now.

**Review by MichaelLueken — APPROVED** [review]
@michelleharrold Tests with different dates are passing without issue.  I have also ran the fundamental WE2E tests on Hera Intel and they all passed.  I will now approve this PR.

**Review by JeffBeck-NOAA — APPROVED** [review]

**Review by willmayfield — APPROVED** [review]

**Comment by MichaelLueken** [comment]
@michelleharrold The automated Jenkins tests passed on Cheyenne, Jet, and Hera.  The WE2E coverage tests were manually run on Orion and all passed.  Moving forward with merging this PR now.

```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
deactivate_tasks                                                   COMPLETE               1.20
get_from_AWS_ics_GEFS_lbcs_GEFS_fmt_grib2_2022040400_ensemble_2me  COMPLETE             755.87
grid_CONUS_3km_GFDLgrid_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta   COMPLETE             257.96
grid_RRFS_AK_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot        COMPLETE             141.27
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_RRFS_v1beta            COMPLETE              14.20
grid_RRFS_CONUS_25km_ics_GSMGFS_lbcs_GSMGFS_suite_GFS_2017_gfdlmp  COMPLETE              10.24
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR              COMPLETE             398.33
grid_RRFS_CONUScompact_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16   COMPLETE              28.01
grid_RRFS_CONUScompact_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16    COMPLETE             279.89
grid_SUBCONUS_Ind_3km_ics_FV3GFS_lbcs_FV3GFS_suite_WoFS_v0         COMPLETE              12.59
nco                                                                COMPLETE               7.64
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE            1907.20
```

**Pull request merged**